File: app/src/main/java/com/example/cardqr/CardAdapter.java
package com.example.cardqr;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.cardqr.model.Card;

import java.util.ArrayList;
import java.util.List;

public class CardAdapter extends RecyclerView.Adapter<CardAdapter.CardViewHolder> {
    private Context context ;

    public CardAdapter(Context context) {
        this.context = context;
    }
    List<Card> cardList = new ArrayList<>();

    public interface OnItemClickListener {
        void onItemClick(Card card);
    }

    private OnItemClickListener itemClickListener;

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.itemClickListener = listener;
    }

    @NonNull
    @Override
    public CardViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_card_qr, parent, false);
        return new CardViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CardViewHolder holder, int position) {
        Card card = cardList.get(position);
        if(card == null){
            return;
        }
        holder.onBindData(card);
    }

    @Override
    public int getItemCount() {
        if(cardList != null){
            return cardList.size();
        }
        return 0;
    }

    public void setData(List<Card> List){
        this.cardList = List;

        notifyDataSetChanged();
    }

    public class CardViewHolder extends RecyclerView.ViewHolder{
        TextView tvNameCard, tvDepartment, tvCompany, tvEmail, tvPhoneNumber, tvWebSite, tvAddress,
                tvFaceBook, tvLinkIn, tvInstagram, tvGithub;
        ImageView ivAvatar;

        public CardViewHolder(@NonNull View itemCard){
            super(itemCard);

            tvNameCard = itemView.findViewById(R.id.tv_name_card);
            tvDepartment = itemView.findViewById(R.id.tv_department);
            tvCompany = itemView.findViewById(R.id.tv_company);
            tvEmail = itemView.findViewById(R.id.tv_email);
            tvPhoneNumber = itemView.findViewById(R.id.tv_phone);
            tvAddress = itemView.findViewById(R.id.tv_address);
            tvWebSite = itemView.findViewById(R.id.tv_website);
            tvFaceBook = itemView.findViewById(R.id.tv_fb);
            tvInstagram = itemView.findViewById(R.id.tv_instagram);
            ivAvatar = itemView.findViewById(R.id.iv_avatar);
            tvLinkIn = itemView.findViewById(R.id.tv_linkin);

        }
        public void onBindData(Card card){
            ivAvatar.setImageURI(Uri.parse(card.getImgAvatar()));
            tvNameCard.setText(card.getNameCard()+"");
            tvDepartment.setText(card.getDepartment());
            tvCompany.setText(card.getCompany());
            tvEmail.setText(card.getEmail());
            tvPhoneNumber.setText(card.getPhoneNumber());
            tvAddress.setText(card.getAddress());
            tvWebSite.setText(card.getWebSite());
            tvFaceBook.setText(card.getFacebook());
            tvInstagram.setText(card.getInstagram());
            tvLinkIn.setText(card.getLinkin());

            setBgr();

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    AlertDialog.Builder dia = new AlertDialog.Builder(v.getContext());
                    dia.setTitle("Dialog Fragment");
                    dia.setIcon(R.drawable.ic_scan_qr);
                    dia.setMessage("ID" + card.getNameCard() + ": " + card.getEmail() + ", "
                            + card.getPhoneNumber());

                    dia.setNegativeButton("No", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                        }
                    });

                    dia.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Toast.makeText(dia.getContext(), "ok", Toast.LENGTH_SHORT).show();
                        }
                    });

                    dia.show();

                    if (itemClickListener != null) {
                        itemClickListener.onItemClick(card);
                    }
                }
            });
        }

        private void setBgr(){
            itemView.setBackgroundResource(R.drawable.bg_white_corner10);
        }
    }
}

File: app/src/main/java/com/example/cardzap/adapters/CardAdapter.kt
package com.example.cardzap.adapters

import android.content.Context
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.RecyclerView
import com.example.cardzap.R
import com.example.cardzap.models.Card

class CardAdapter(private val context: Context) : RecyclerView.Adapter<CardAdapter.CardViewHolder>() {
    private var cardList: List<Card> = ArrayList()
    private var itemClickListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(card: Card)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        itemClickListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_card, parent, false)
        return CardViewHolder(view)
    }

    override fun onBindViewHolder(holder: CardViewHolder, position: Int) {
        val card = cardList[position] ?: return
        holder.onBindData(card)
    }

    override fun getItemCount(): Int {
        return cardList.size
    }

    fun setData(list: List<Card>) {
        cardList = list
        notifyDataSetChanged()
    }

    inner class CardViewHolder(itemCard: View) : RecyclerView.ViewHolder(itemCard) {
        private val tvNameCard: TextView = itemView.findViewById(R.id.tv_name_card)
        private val tvDepartment: TextView = itemView.findViewById(R.id.tv_department)
        private val tvCompany: TextView = itemView.findViewById(R.id.tv_company)
        private val tvEmail: TextView = itemView.findViewById(R.id.tv_email)
        private val tvPhoneNumber: TextView = itemView.findViewById(R.id.tv_phone)
        private val tvAddress: TextView = itemView.findViewById(R.id.tv_address)
        private val tvWebSite: TextView = itemView.findViewById(R.id.tv_website)
        private val tvFaceBook: TextView = itemView.findViewById(R.id.tv_fb)
        private val ivAvatar: ImageView = itemView.findViewById(R.id.iv_avatar)
        private val tvLinkIn: TextView = itemView.findViewById(R.id.tv_linkin)

        fun onBindData(card: Card) {
            ivAvatar.setImageURI(Uri.parse(card.imgAvatar))
            tvNameCard.text = card.nameCard
            tvDepartment.text = card.department
            tvCompany.text = card.company
            tvEmail.text = card.email
            tvPhoneNumber.text = card.phoneNumber
            tvAddress.text = card.address
            tvWebSite.text = card.webSite
            tvFaceBook.text = card.facebook
            tvLinkIn.text = card.linkin

            setBgr()

            itemView.setOnClickListener {
                val dia = AlertDialog.Builder(it.context)
                dia.setTitle("Dialog Fragment")
                dia.setIcon(R.mipmap.ic_scan_qr)
                dia.setMessage("ID ${card.nameCard}: ${card.email}, ${card.phoneNumber}")

                dia.setNegativeButton("No") { _, _ ->
                }

                dia.setPositiveButton("Yes") { _, _ ->
                    Toast.makeText(dia.context, "ok", Toast.LENGTH_SHORT).show()
                }

                dia.show()

                itemClickListener?.onItemClick(card)
            }
        }

        private fun setBgr() {
            itemView.setBackgroundResource(R.drawable.bg_white_corner10)
        }
    }
}